frontend/apps/picasso/stores/defi/polkadot/transfers/subscribers/transactionFee.ts

```diff
@@ -0,0 +1,68 @@
+import { AllProviders } from "@/defi/polkadot/context/hooks";
+import { Executor, getSigner } from "substrate-react";
+import { useStore } from "@/stores/root";
+import { APP_NAME } from "@/defi/polkadot/constants";
+import { fromChainIdUnit, unwrapNumberOrHex } from "shared";
+import BigNumber from "bignumber.js";
+
+export const subscribeTransactionFee = async (
+  allProviders: AllProviders,
+  walletAddress: string,
+  executor: Executor
+) => {
+  return useStore.subscribe(
+    (store) => ({
+      isLoaded: store.substrateTokens.isLoaded,
+      selectedRecipient: store.transfers.recipients.selected,
+      sourceChain: store.transfers.networks.from,
+      transferExtrinsic: store.transfers.transferExtrinsic,
+    }),
+    async ({ selectedRecipient, sourceChain, isLoaded, transferExtrinsic }) => {
+      if (!isLoaded || !transferExtrinsic) return;
+      const recipient = selectedRecipient.length
+        ? selectedRecipient
+        : walletAddress;
+
+      const api = allProviders[sourceChain].parachainApi;
+      if (!api) return;
+
+      const call = useStore
+        .getState()
+        .transfers.makeTransferCall(api, recipient);
+      if (!call) return;
+
+      const signer = await getSigner(APP_NAME, walletAddress);
```

Comment:
@easteregg sir its better to use the `useSigner` hook, `getSigner` is not hooked up with original context

Reply:
Not possible, this is outside react component ecosystem. 

Once we move out from context, we can reach the store directly.